packages/debug/src/index.ts

```diff
@@ -1,88 +1,174 @@
 
-const MAX_LOGS = 100
+const argsHistory: [namespace: string, ...unknown[]][] = []
+let lastTimestamp = Date.now()
+let lastColor = 0
 
-const debugArgsHistory: any[] = []
+globalThis.DEBUG ??= process.env.DEBUG ?? ''
+globalThis.DEBUG_COLORS ??= process.env.DEBUG_COLORS ? process.env.DEBUG_COLORS === 'true' : true
 
-// Patch the Node.js logger to use `console.debug` or `console.log` (similar to
-// the browser logger) in the Edge Client.
-if (typeof process !== 'undefined' && typeof process.stderr?.write !== 'function') {
-  debug.log = console.debug ?? console.log
+/**
+ * Top-level utilities to configure the debug module.
+ *
+ * @example
+ * ```ts
+ * import Debug from '@prisma/debug'
+ * Debug.enable('prisma:client')
+ * const debug = Debug('prisma:client')
+ * debug('Hello World')
+ * ```
+ */
+const topProps = {
+  enable(namespace: any) {
+    if (typeof namespace === 'string') {
+      globalThis.DEBUG = namespace
+    }
+  },
+  disable() {
+    const prev = globalThis.DEBUG
+    globalThis.DEBUG = ''
+    return prev
+  },
+  // this is the core logic to check if logging should happen or not
+  enabled(namespace: string) {
+    // these are the namespaces that we are listening to in DEBUG=...
+    const listenedNamespaces: string[] = globalThis.DEBUG.split(',')
+
+    // we take incoming namespaces and check then against listened
+    return listenedNamespaces.every((listenedNamespace) => {
+      // namespaces can be negated with a `-` prefix (exclusions)
+      const isNegated = listenedNamespace.startsWith('-') ? 1 : 0
+      listenedNamespace = listenedNamespace.slice(isNegated)
+
+      // we split the namespaces by `:` to be able to compare them
+      let emittedParts = namespace.split(':')
+      let listenedParts = listenedNamespace.split(':')
+      const sizeDiff = listenedParts.length - emittedParts.length
+
+      // here we make that the two compared arrays have the same length
+      if (listenedParts.length > emittedParts.length) {
+        emittedParts = [...emittedParts, ...Array(Math.abs(sizeDiff)).fill(undefined)]
+      } else if (listenedParts.length < emittedParts.length) {
+        // if the last part is a wildcard, we fill the rest with wildcards
+        const filler = listenedParts[listenedParts.length - 1] === '*' ? '*' : undefined
+        listenedParts = [...listenedParts, ...Array(Math.abs(sizeDiff)).fill(filler)]
+      }
+
+      // matches if each part is equal or if the listened part is a wildcard
+      const matched = listenedParts.every((listenedPart, i) => {
+        return listenedPart === emittedParts[i] || (emittedParts[i] && listenedPart === '*')
+      })
+
+      return isNegated ? !matched : matched // flip the result if negated
+    })
+  },
+  log: (...args: string[]) => {
+    const [ns, format, ...rest] = args
+    // concat `ns`+`format` because console only formats first arg
+    console.warn(`${ns} ${format}`, ...rest)
```

Comment:
```suggestion
    console.log(`${ns} ${format}`, ...rest)
```

Reply:
Actually, I just changed it from `log` to `warn` because `debug` does push to stderr, not stdout.

Reply:
Right by default debug will log to stderr